### ISSUE TITLE
Provide Spring-RestTemplate integration

### ISSUE THREAD

**Issue by imasahiro** [line/armeria#989]
[RestTemplate](https://github.com/spring-projects/spring-framework/blob/master/spring-web/src/main/java/org/springframework/web/client/RestTemplate.java) is a spring feature that provides (a)synchronous client-side HTTP access. And it can replace httpclient implementation via ClientHttpRequestFactory (e.g. [Apache HttpClient](https://github.com/spring-projects/spring-framework/blob/873cb4e58b5103c2e901ceaf75fad38f4edc5c82/spring-web/src/main/java/org/springframework/http/client/HttpComponentsClientHttpRequest.java), [OkHttp](https://github.com/spring-projects/spring-framework/blob/873cb4e58b5103c2e901ceaf75fad38f4edc5c82/spring-web/src/main/java/org/springframework/http/client/OkHttp3ClientHttpRequestFactory.java), [Netty4](https://github.com/spring-projects/spring-framework/blob/873cb4e58b5103c2e901ceaf75fad38f4edc5c82/spring-web/src/main/java/org/springframework/http/client/Netty4ClientHttpRequestFactory.java)).

Think it is good idea to support Armeria's httpclient. How do you think?

**Comment by trustin** [comment]
Sounds good to me! Are you up for this?

**Comment by imasahiro** [comment]
Yep, I'm trying to make it based on Netty4ClientHttpRequestFactory

**Comment by trustin** [comment]
Very cool. How about adding `armeria-spring` to put Spring framework -related stuff?

**Comment by imasahiro** [comment]
Yeah, will want to reorg spring-boot dir and spring FW stuff like this.
* spring/spring-boot-autoconfigure
* spring/spring-boot-starter
* spring/spring-resttemplate
How do you think? Just add spring dir is enough..?

**Comment by trustin** [comment]
> - spring/spring-boot-autoconfigure
> - spring/spring-boot-starter
> - spring/spring-resttemplate

How about:

- spring/boot/autoconfigure
- spring/boot/starter
- spring/resttemplate (or spring/web, spring/webmvc, ...)

>  How do you think? Just add spring dir is enough..?

I think it depends on which Spring artifact REST template belongs to. For example, if it belongs to spring-webmvc, then we could put it into `armeria-spring-webmvc`.

**Comment by eonezhang** [comment]
I think resttemplate should belong to client, how about like okhttp?
